test/e2e/publish_confirm.test.ts

```diff
@@ -0,0 +1,107 @@
+import { expect } from "chai"
+import { randomUUID } from "crypto"
+import { connect, Connection } from "../../src"
+import { Rabbit } from "../support/rabbit"
+import { eventually } from "../support/util"
+
+describe("publish a message and get confirmation", () => {
+  const rabbit = new Rabbit()
+  let connection: Connection
+
+  beforeEach(async () => {
+    connection = await connect({
+      hostname: "localhost",
+      port: 5552,
+      username: "rabbit",
+      password: "rabbit",
+      vhost: "/",
+      frameMax: 0, // not used
+      heartbeat: 0, // not used
+    })
+  })
+  afterEach(() => connection.close())
+  afterEach(() => rabbit.closeAllConnections())
+
+  it("after the server replies with a confirm, the confirm callback is invoked", async () => {
+    const stream = `my-stream-${randomUUID()}`
+    await rabbit.createStream(stream)
+    let confirmed = false
+    const publisher = await connection.declarePublisher({
+      stream,
+      publisherRef: "my publisher",
+      confirmCb: () => {
```

Comment:
Instead of pass the callback in the declaration, I would expect something more classic like:
```typescript
const publisher = await connection.declarePublisher(...)
publish.on("confirm", (error, publishingId) => { ...})